memgpt/schemas/block.py

```diff
@@ -0,0 +1,51 @@
+from typing import List, Optional, Union
+
+from pydantic import BaseModel, Field, validator
+
+# block of the LLM context
+
+
+class Block(BaseModel):
+    """Block of the LLM context"""
+
+    description: Optional[str] = Field(None, description="Description of the block.")
+    limit: int = Field(2000, description="Character limit of the block.")
+    value: Optional[Union[List[str], str]] = Field(None, description="Value of the block.")
+
+    def __setattr__(self, name, value):
```

Comment:
there is a built-in for this in pydantic, `validate_assignment` https://docs.pydantic.dev/latest/api/config/#pydantic.config.ConfigDict.validate_assignment